doc/manual/syntax.md

```diff
@@ -567,3 +569,35 @@ Examples:
 > {foo | priority -1 = 1} & {foo = 2}
 { foo = 2 }
 ```
+
+The `optional` metadata indicates that a field is not mandatory, and is mostly
+used inside record contracts:
+
+```text
+> let Contract = {
+    foo | Num,
+    bar | Num
+        | optional,
+  }
+> let value | Contract = {foo = 1}
+> value
+{ foo = 1 }
+
+> {bar = 1} | Contract
+error: missing definition for `foo`
+[..]
+```
+
+The `not_exported` metadata indicates that a field shouldn't appear in the
+serialization (including the output of the `nickel export` command):
+
+```text
+> let value = { foo = 1, bar | not_exported = 2}
+> value
+{ foo = 1, bar = 2 }
+
+> builtin.serialize `Json value
+"{
+  "foo": 1
+}"
+```
```

Comment:
The metadata section of the manual might need a more comprehensive overhaul, since we now have a distinction between `|` at record fields and `|` as contract assertion for values. But that's probably better done in another PR down the line.

Reply:
Good point. Let's open an issue for that.